src/test/java/seedu/jarvis/model/cca/CcaListTest.java

```diff
@@ -96,10 +96,29 @@ public void setCca_editedCcaHasDifferentIdentity_success() {
         assertEquals(expectedCcaList, ccaList);
     }
 
+    // TODO fix this test.
     //    @Test
     //    public void setCca_editedCcaHasNonUniqueIdentity_throwsDuplicateCcaException() {
     //        ccaList.addCca(CANOEING);
     //        ccaList.addCca(GUITAR_ENSEMBLE);
     //        assertThrows(DuplicateCcaException.class, () -> ccaList.updateCca(CANOEING, GUITAR_ENSEMBLE));
     //    }
+
+    @Test
+    public void remove_nullCca_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ccaList.removeCca(null));
+    }
+
+    @Test
+    public void remove_ccaDoesNotExist_throwsCcaNotFoundException() {
+        assertThrows(CcaNotFoundException.class, () -> ccaList.removeCca(CANOEING));
+    }
+
+    @Test
+    public void remove_existingCca_removesCca() {
```

Comment:
you should do another test that removes a cca in the middle so that you know you're removing the cca at the correct index - zero based and one based numbers are tricky